### DISCUSSION TITLE
@ioc:Adonis meaning

### DISCUSSION THREAD

**Discussion by smuralimohan** [adonisjs/core#3455] [General]
Hi, 

What does the prefix '@ioc:Adonis' mean in import definitions? My jest tests are unable to resolve the modules imported in this fashion. What is the resolution mechanism for such imports and where can I read more about this syntax? 

Thanks,
Murali

**Comment by smuralimohan** [comment]
Can someone let me know if there is any alternative import syntax for 
import Hash from '@ioc:Adonis/Core/Hash'
without the @ioc:Adonis prefix?

**Comment by smuralimohan** [comment]
Can't get an answer for such a simple thing... suxxxx :(

**Comment by thetutlage** [comment]
Lol. Even paid customer support doesn't reply under 6 hours and here we have someone getting upset that why the query is un-answered on an open source project.

Practice patience.

Same for this https://github.com/adonisjs/core/discussions/3120#discussioncomment-1886418

**Reply by smuralimohan** [reply]
This question was lingering for months, buddy. It is not just hours. And I still haven't got a valid response... but an advice to practice patience.... for what months?
